Hmmmm, interesting. Pulled fergus/ffi branch and got the following on make. 
```
Undefined symbols for architecture arm64:
  "_tbabs_", referenced from:
      _genreltrans_ in wrappers.o
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
make: *** [build/lib/libreltrans.dylib] Error 1
```
This is linked specifically to the referring external name as "tbabs_" in R8 of `genreltrans.f90.` Removing the trailing underscore successfully compiles. I think @fjebaker's idea of attaching to `C_tbabs` is probably wise, otherwise we need a mac and linux version of the code (or need a way to know this on compilation and switch out the names accordingly).

--- a/subroutines/amodules.f90
+++ b/subroutines/amodules.f90
@@ -50,7 +50,20 @@ MODULE dyn_gr
 END MODULE dyn_gr
 
 module xillver_tables
-    implicit none 
+    implicit none
+
+    interface
+        ! An interface to the xsatbl function in libXSFunctions.
+        subroutine xsatbl(ear,ne,params,filename,ifl,photar,photer)          &
+            bind(C, name="xsatbl")
+            use iso_c_binding, only: c_float, c_int, c_char
+            real(c_float), dimension(*), intent(in) :: ear, params
+            real(c_float), dimension(*), intent(out) :: photar, photer
+            character(kind=c_char), intent(in) :: filename(*)
+            integer(c_int), value, intent(in) :: ne, ifl
+        end subroutine xsatbl
+    end interface
+
     character (len=50), parameter ::  xillver = 'xillver-a-Ec5_normalised.fits'
     character (len=50), parameter ::  xillverDCp = 'xillverCp_v3.4_normalised.fits'
     character (len=200) ::  path_tables 

--- a/subroutines/genreltrans.f90
+++ b/subroutines/genreltrans.f90
@@ -22,7 +22,7 @@ contains
     !
     ! This wrapper temporarily performs a runtime cast on all of the arrays to
     ! double precision, as the C_tbabs function that we will eventually call
-    ! expects double precision, whilst much of reltrans still uses `real`, wihch
+    ! expects double precision, whilst much of reltrans still uses `real`, which
     ! maps to single precision.
     !
     ! Once the precision has been modified, this function can be simplified.

--- a/subroutines/genreltrans.f90
+++ b/subroutines/genreltrans.f90
@@ -2,7 +2,51 @@ module m_genreltrans
     use common_types
     implicit none
 
+    interface
+        ! The interface around the XSPEC C_tbabs function.
+        ! It will call the C_tbabs symbol in the libXSFunctions shared library.
+        subroutine c_tbabs(earx, nex, params, Ifl, absorbx, photerx, str)      &
+            bind(C, name = "C_tbabs")
+            use iso_c_binding, only: c_double, c_int, c_char
+            real(c_double), intent(in) :: earx(nex+1)
+            real(c_double), intent(in) :: params(1)
+            real(c_double), intent(inout) :: absorbx(nex), photerx(nex)
+            integer(c_int), value, intent(in) :: nex, Ifl
+            character(kind=c_char), intent(in) :: str(*)
+        end subroutine c_tbabs
+    end interface
+
 contains
+
+    ! Call the tbabs function from the XSPEC model library.
+    !
+    ! This wrapper temporarily performs a runtime cast on all of the arrays to
+    ! double precision, as the C_tbabs function that we will eventually call
+    ! expects double precision, whilst much of reltrans still uses `real`, wihch
+    ! maps to single precision.
+    !
+    ! Once the precision has been modified, this function can be simplified.
+    subroutine tbabs(earx, nex, nh, Ifl, absorbx, photerx)
+        real, intent(in) :: earx(0:nex), nh
+        real, intent(inout) :: absorbx(nex), photerx(nex)
+        integer, intent(in) :: nex, Ifl
+
+        double precision :: d_earx(0:nex), d_absorbx(nex), d_photerx(nex),     &
+            d_params(1)
+        integer i
+
+        do i = 0,nex
+            d_earx(i) = earx(i)
+        end do
+
+        d_params(1) = nh
+        call c_tbabs(d_earx,nex,d_params,Ifl,d_absorbx,d_photerx, "")
+
+        do i = 1,nex
+            absorbx(i) = d_absorbx(i)
+            photerx(i) = d_photerx(i)
+        end do
+    end subroutine tbabs
 
     ! Allocate the global arrays that reltrans needs
     subroutine setup_global_arrays(config, nlp)

--- a/subroutines/rest_frame_reflection/get_reflionx.f90
+++ b/subroutines/rest_frame_reflection/get_reflionx.f90
@@ -20,12 +20,14 @@ subroutine get_reflionx(ear, ne, param, ifl, photar)
         write(*,*)"Enter reflionx file (with full path)"
         read(*,'(a)')filenm
      end if
-     path_name_reflionx_table = trim(filenm) 
+     path_name_reflionx_table = trim(filenm)
      write(*,*) 'Set the reflionx table at ', path_name_reflionx_table
      needfile = .false.
   end if
 ! Interpolate a spectrum from it
-  call xsatbl(ear, ne, param,  path_name_reflionx_table, ifl, photar, photer) 
+! Pass the null terminator for C compatability, as xsatbl is an external C function.
+  call xsatbl(ear, ne, param,  [trim(path_name_reflionx_table), char(0)], ifl, &
+    photar, photer)
   return
 end subroutine get_reflionx
 !-----------------------------------------------------------------------

--- a/subroutines/rest_frame_reflection/get_xillver.f90
+++ b/subroutines/rest_frame_reflection/get_xillver.f90
@@ -25,9 +25,15 @@
 
       ifl = 0
       if( Cp .eq. -1 )then         !xillver
-         call xsatbl(ear, ne, param_xillPL, trim(pathname_xillver), ifl, photar, photer)
+         ! Pass the null terminator for C compatability, as xsatbl is an
+         ! external C function.
+         call xsatbl(ear, ne, param_xillPL,                                    &
+            [trim(pathname_xillver), char(0)], ifl, photar, photer)
       else if ( Cp .eq. 2 )then    !xillverDCp
-         call xsatbl(ear, ne, param_xillCp, trim(pathname_xillverDCp), ifl, photar, photer)
+         ! Pass the null terminator for C compatability, as xsatbl is an
+         ! external C function.
+         call xsatbl(ear, ne, param_xillCp,                                    &
+            [trim(pathname_xillverDCp), char(0)], ifl, photar, photer)
       else
          write(*,*) 'No xillver model available for this configuration'
          stop 

